glean-core/ios/GleanTests/UploaderCapabilityTests.swift

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@testable import Glean
+import XCTest
+
+// REASON: This test is long because of setup boilerplate
+class UploaderCapabilityTests: XCTestCase {
+    var expectation: XCTestExpectation?
+
+    private let testRequest = PingRequest(
+        documentId: "12345",
+        path: "/some/random/path/not/important",
+        body: [UInt8]("{ \"ping\": \"test\" }".utf8),
+        headers: [:],
+        bodyHasInfoSections: true,
+        pingName: "testPing",
+        uploaderCapabilities: []
+    )
+
+    override func setUp() {
+        resetGleanDiscardingInitialPings(testCase: self, tag: "UploaderCapabilityTests", clearStores: true)
+    }
+
+    override func tearDown() {
+        // Reset expectations
+        expectation = nil
+        tearDownStubs()
+    }
+
+    /// Launch a new ping uploader on the background thread.
+    ///
+    /// This function doesn't block.
+    private func getUploader(_ capabilities: [String] = []) -> HttpPingUploader {
+        // Build a URLSession with no-caching suitable for uploading our pings
+        let config: URLSessionConfiguration = .default
+        config.requestCachePolicy = NSURLRequest.CachePolicy.reloadIgnoringLocalCacheData
+        config.urlCache = nil
+        let session = URLSession(configuration: config)
+        return HttpPingUploader.init(configuration: Configuration(), session: session, capabilities: capabilities)
+    }
+
+    func testUploaderIncapable() {
+        var request = testRequest
+        request.uploaderCapabilities = ["ohttp"]
+
+        let httpPingUploader = self.getUploader()
+        httpPingUploader.upload(request: request) { result in
+            XCTAssertEqual(
+                .incapable(unused: 0),
+                result,
+                "upload should return .incapable when capabilities don't match"
+            )
+        }
+    }
+
+    func testUploaderCapable() {
+        var request = testRequest
+        request.uploaderCapabilities = ["ohttp"]
```

Comment:
Seems like, since everywhere you use `testRequest` you're giving it the sample `uploaderCapabilities` that we could instead set the `uploaderCapabilities: ["ohttp"]` in its decl and call it `ohttpRequest`.

Reply:
We could, these tests were originally just grouped with the other HttpUploaderTests and so I had made a way to pass capabilities as needed. Since Xcode so nicely forced me to move these to their own test class due to some conflict with unrelated tests, that flexibility is no longer needed.

Reply:
Let me fix that up right quick